frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/ToggleListingCommitsAction.java

```diff
@@ -38,16 +39,30 @@ public void update(AnActionEvent anActionEvent) {
     if (branchLayout.isDefined()) {
       boolean anyChildBranchExists = branchLayout.get().getRootEntries()
           .exists(rootBranch -> rootBranch.getChildren().nonEmpty());
+
       if (anyChildBranchExists) {
-        presentation.setEnabled(true);
-        presentation.setDescription("Toggle listing commits");
+        boolean anyCommitExists = getGitMacheteRepositorySnapshot(anActionEvent)
+            .map(repo -> repo.getRootBranches()
+                .flatMap(root -> root.getDownstreamBranches())
+                .exists(b -> b.getCommits().nonEmpty()))
+            .getOrElse(false);
+
+        if (anyCommitExists) {
+          presentation.setEnabled(true);
+          presentation.setDescription(GitMacheteBundle.message("action.toggle-listing-commits.description"));
+        } else {
+          presentation.setEnabled(false);
+          presentation
+              .setDescription(GitMacheteBundle.message("action.toggle-listing_commits.description.disabled.no-commits"));
+        }
       } else {
         presentation.setEnabled(false);
-        presentation.setDescription("Toggle listing commits disabled: no child branches present");
+        presentation
+            .setDescription(GitMacheteBundle.message("action.toggle-listing-commits.description.disabled.no-child-branches"));
       }
     } else {
```

Comment:
Maybe rework nested if-elses into if-returns to keep each piece of error handling closer to its corresponding condition